lib/ansible/cli/doc.py

```diff
@@ -948,30 +958,20 @@ def print_paths(finder):
         return os.pathsep.join(ret)
 
     @staticmethod
-    def get_snippet_text(doc):
+    def get_snippet_text(doc, pytpe='module'):
 
         text = []
-        desc = DocCLI.tty_ify(doc['short_description'])
-        text.append("- name: %s" % (desc))
-        text.append("  %s:" % (doc['module']))
-        pad = 31
-        subdent = " " * pad
-        limit = display.columns - pad
-
-        for o in sorted(doc['options'].keys()):
-            opt = doc['options'][o]
-            if isinstance(opt['description'], string_types):
-                desc = DocCLI.tty_ify(opt['description'])
-            else:
-                desc = DocCLI.tty_ify(" ".join(opt['description']))
 
-            required = opt.get('required', False)
-            if not isinstance(required, bool):
-                raise("Incorrect value for 'Required', a boolean is needed.: %s" % required)
-            if required:
-                desc = "(required) %s" % desc
-            o = '%s:' % o
-            text.append("      %-20s   # %s" % (o, textwrap.fill(desc, limit, subsequent_indent=subdent)))
+        if ptype == 'lookup':
+            _do_lookup_snippet(text, doc)
+        elif ptype == 'module':
+            _do_yaml_snippet(text, doc)
+        elif 'options' in doc:
+            _do_ini_snippet(text, doc)
+        else:
```

Comment:
This should probably be an error condition (or print the man text as the previous version did but an error feels like it's more informative).

Reply:
Not sure it warrants an error, a warning perhaps?

This can be the result of the specific plugin having no options, as used to be the case for script inventory.